src/rebar3_hex_docs.erl

```diff
@@ -156,6 +157,32 @@ resolve_doc_dir(AppInfo) ->
     Dir = proplists:get_value(dir, EdocOpts, ?DEFAULT_DOC_DIR),
     proplists:get_value(doc, AppDetails, Dir).
 
+maybe_gen_docs(State) -> 
+    case doc_opts(State) of
+        undefined -> 
+             rebar_api:error("No docs config found", []),
+            noop;
+        PrvName -> 
+            AllProviders = rebar_state:providers(State),
+             case providers:get_provider(PrvName, AllProviders) of
+                 not_found -> 
+                     rebar_api:error("No provider found for ~ts", [PrvName]),
+                     noop;
+                 Prv ->
+                     providers:do(Prv, State)
+             end
+    end.
+
+doc_opts(State) -> 
+    Opts = rebar_state:opts(State),
```

Comment:
Maybe need to get this info from `App`vs `State` otherwise we could be operating on the wrong app 🤔

Reply:
That's a fair concern. That being said if you're publishing to hex the assumption is that you're in an app-specific directory so in practice it shouldn't be a problem. If you're in an umbrella app you shouldn't be able to publish that easily right? Usually you `cd` into the app dir and do your thing there.

Reply:
Yeah, we actually removed the support for publishing umbrellas iirc too, so we're good 👍